.github/workflows/test.yml

```diff
@@ -0,0 +1,68 @@
+name: test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: linux
+            runs-on: ubuntu-24.04
+            arch: x86_64
+
+          - os: macOS
+            runs-on: macos-14
+            arch: aarch64
+
+    name: "${{ matrix.os }}-${{ matrix.arch }}"
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # 4.1.7
+
+      - name: On Linux, switch to GCC 14
+        if: runner.os == 'Linux'
+        run: |
+          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-14 100
+          sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-14 100
+          gcc --version
+
+      - name: On macOS, allow Meson to find libcrypto and libssl
+        if: runner.os == 'macOS'
+        run: |
+          # For now, symlink rather than altering `meson.build` to pass
+          # `dirs : ['/opt/homebrew/lib']` to the relevant find_library()` calls.
+          sudo mkdir -p /usr/local/lib
+          sudo ln -s /opt/homebrew/lib/{libcrypto,libssl}.dylib /usr/local/lib/
+
+      - name: Install Meson and Ninja
+        shell: bash
+        run: |
+          case "${RUNNER_OS}" in
+            Linux)
+              sudo apt install meson ninja-build
+              ;;
+            macOS)
+              brew install meson ninja
+              ;;
+            *)
+              exit 1
```

Comment:
lets use github expressions for this:

```
- name: install deps (linus)
  if: runner.os == 'Linux'
  run: sudo apt install meson ninja build

- name: install deps (mac)
  if: runner.os == 'macOS'
  ...
```

Reply:
When split it just shows up better in GitHub logs. The command you actually care about is directly displayed vs being collapsed